src/screens/VisualTests/BuildResults.tsx

```diff
@@ -154,20 +159,22 @@ export const BuildResults = ({
       </Section>
       <Section>
         <Bar>
-          <Col>
-            <WithTooltip
-              tooltip={<TooltipNote note="Switch snapshot" />}
-              trigger="hover"
-              hasChrome={false}
-            >
-              <IconButton
-                data-testid="button-toggle-snapshot"
-                onClick={() => toggleBaselineImage()}
+          {!completedNoChanges && (
+            <Col>
+              <WithTooltip
+                tooltip={<TooltipNote note="Switch snapshot" />}
+                trigger="hover"
+                hasChrome={false}
               >
-                <Icons icon="transfer" />
-              </IconButton>
-            </WithTooltip>
-          </Col>
+                <IconButton
+                  data-testid="button-toggle-snapshot"
+                  onClick={() => toggleBaselineImage()}
```

Comment:
Do we need some logic in case you

a) select to show the baseline image
b) you switch to a story with no changes

Or does the state variable reset each time you change story?

Reply:
@tmeasday thank you for checking this. I looked at it, and the baselineImageVisible value does not reset when changing stories because the SnapshotComparison component is never unmounted. I've added some logic to reset the state of all the state variables if the storyId changes (added it as a prop as well). It works, but there is a small flicker when it resets but the new story hasn't loaded yet.